jwst/imprint/imprint_step.py

```diff
@@ -19,46 +19,46 @@ class ImprintStep(Step):
 
     def process(self, input, imprint):
 
-        # subtract leakcal (imprint) image
-        # If there is only one imprint image is in the association use it  for all the data.
-        # If there is more than one imprint image in the association then select
-        # the imprint  image to subtract based on position number (DataModel.meta.dither.position_number) &
-        # observation number (Datamodel.meta.observation.observation_number)
-
-        # Open the input data model and get position number of image
+        # Open the input science image and get its dither pattern position number
         input_model = datamodels.open(input)
         pos_no = input_model.meta.dither.position_number
         obs_no = input_model.meta.observation.observation_number
 
-        # find imprint that goes with input image - if there is only 1 imprint image - use it.
+        # If there is only one imprint image listed in the association,
+        # use it for all science images.
+        # If there is more than one imprint image in the association,
+        # then select the imprint image to be used by matching the dither pattern position
+        # number and observation number to the science image.
         num_imprint = len(imprint)
-        match = None
         if num_imprint == 1:
-            match = 0
+            match = 0  # use the first, and only, imprint image
         else:
-            i = 0
-            while match is None and i < num_imprint:
-                # open imprint images and read in pos_no to find match
+            match = None
+            for i in range(num_imprint):
+                # open imprint image and read in pos_no to find match
                 imprint_model = datamodels.open(imprint[i])
                 imprint_pos_no = imprint_model.meta.dither.position_number
                 imprint_obs_no = imprint_model.meta.observation.observation_number
                 imprint_model.close()
                 if pos_no == imprint_pos_no and obs_no == imprint_obs_no:
                     match = i
-                i = i + 1
+                    break
 
-        # Initialize result to be input (just in case no matching imprint image was found)
+        # Copy the input image to the output (just in case no matching imprint image was found)
         result = input_model.copy()
 
         if match is not None:
+            # Subtract the matching imprint image
+            self.log.info(f"Subtracting imprint image {imprint[match]}")
```

Comment:
Have you had a chance to view this output, say from both CLI `strun` commands and from notebook calls? I'm curious what is returned if the list of imprints is a list of filenames versus datamodels - if datamodels, you may get something looking more like `'<ImageModel(8, 8) from jw01163002001_03102_00001_nrca3_rate.fits>'`. Could consider logging post-open, then logging the model.meta.filename.

Reply:
Yes, you're right that I currently see a log message of the form "<IFUImageModel(2048, 2048) from jw_blah.fits". I'll go ahead and make your suggested change. Good idea.